deserializeMsgPack() Filtering support

imwhocodes opened the pull request:
This pull request add filtering support for MsgPack deserialization

Based on deserializeJson() code/logic

This commit passed current unit testing.
No breaking change or regression
Still missing unit test for specific new feature (filtered MsgPack deserialization)

imwhocodes committed:
MsgPack Filter, created skipping consumer, WIP

imwhocodes committed:
Added filter logic, WIP

imwhocodes committed:
unit tested

bblanchon commented:
Hi @imwhocodes,

Thank you very much for this PR, but as I told you, after seeing your message, I started working on my implementation.
I'm sorry you had to do it on your own; again, I didn't realize how bad you needed this feature.
I cannot use your contribution because you didn't write any unit tests and writing them afterward is the best way to miss many code paths.
I'll leave this PR opened until I merge mine and mention your name in the release notes.

Best regards,
Benoit

imwhocodes committed:
added messagepack unit test, fixed filtering bug

imwhocodes commented:
Don't worry, I get it,
but I needed it ASAP so I keept working on it, **today I added unit testing** (and founded and fixed some bugs)

If you want check if the test that I implemented are enough (ported from the json one, skipped the one testing for syntax and comments)

Do as you wish,
thanks anyway

imwhocodes committed:
Delete settings.json

Remove wrongly included vscode setting

imwhocodes committed:
added unit test for numeric and string codepath, enable skipping of LONG_LONG even if not supported

imwhocodes committed:
Merge branch '6.x' of https://github.com/imwhocodes/ArduinoJson into 6.x

imwhocodes committed:
added autogenerated test for root array and objects
